Sources/Sentry/SentryMigrateSessionInit.m

```diff
@@ -0,0 +1,117 @@
+#import <Foundation/Foundation.h>
+
+#import "SentryEnvelope.h"
+#import "SentryEnvelopeItemType.h"
+#import "SentryLog.h"
+#import "SentryMigrateSessionInit.h"
+#import "SentrySerialization.h"
+#import "SentrySession+Private.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SentryMigrateSessionInit
+
++ (void)migrateSessionInit:(NSString *)envelopeFilePath
+          envelopesDirPath:(NSString *)envelopesDirPath
+         envelopeFilePaths:(NSArray<NSString *> *)envelopeFilePaths;
+{
+    NSData *envelopeData = [[NSFileManager defaultManager] contentsAtPath:envelopeFilePath];
+    SentryEnvelope *envelope = [SentrySerialization envelopeWithData:envelopeData];
+    if (nil == envelope) {
+        return;
+    }
+
+    for (SentryEnvelopeItem *item in envelope.items) {
+        if ([item.header.type isEqualToString:SentryEnvelopeItemTypeSession]) {
+            SentrySession *session = [SentrySerialization sessionWithData:item.data];
+            if (nil != session && [session.flagInit boolValue]) {
+                [self setInitFlagOnNextEnvelopWithSameSessionId:session
+                                               envelopesDirPath:envelopesDirPath
+                                              envelopeFilePaths:envelopeFilePaths];
+            }
+        }
+    }
+}
+
++ (void)setInitFlagOnNextEnvelopWithSameSessionId:(SentrySession *)session
+                                 envelopesDirPath:(NSString *)envelopesDirPath
+                                envelopeFilePaths:(NSArray<NSString *> *)envelopeFilePaths
+{
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+
+    for (NSString *envelopeFilePath in envelopeFilePaths) {
+        NSString *envelopePath = [envelopesDirPath stringByAppendingPathComponent:envelopeFilePath];
+        NSData *envelopeData = [fileManager contentsAtPath:envelopePath];
+        SentryEnvelope *envelope = [SentrySerialization envelopeWithData:envelopeData];
+
+        if (nil != envelope) {
+            [self setInitFlagIfContainsSameSessionId:session.sessionId
+                                            envelope:envelope
+                                    envelopeFilePath:envelopePath];
+        }
+    }
+}
+
++ (void)setInitFlagIfContainsSameSessionId:(NSUUID *)sessionId
+                                  envelope:(SentryEnvelope *)envelope
+                          envelopeFilePath:(NSString *)envelopeFilePath
+{
+    for (SentryEnvelopeItem *item in envelope.items) {
+        if ([item.header.type isEqualToString:SentryEnvelopeItemTypeSession]) {
+            SentrySession *localSession = [SentrySerialization sessionWithData:item.data];
+
+            if (nil != localSession && [localSession.sessionId isEqual:sessionId]) {
+                [localSession setFlagInit];
+
+                [self storeSessionInit:envelope session:localSession path:envelopeFilePath];
```

Comment:
```suggestion
                [self storeSessionInit:envelope session:localSession path:envelopeFilePath];
                break;
```

Reply:
Good catch. This makes the code faster and avoids moving the init flag to multiple session updates.